src/clr-angular/data/datagrid/providers/selection.spec.ts

```diff
@@ -131,6 +139,13 @@ export default function(): void {
         expect(selectionInstance.isAllSelected()).toBe(true);
       });
 
+      it('should return true when all posible items are selected excluding locked', function() {
+        selectionInstance.selectionType = SelectionType.Multi;
+        selectionInstance.lockItem(4, true);
+        selectionInstance.toggleAll();
+        expect(selectionInstance.isAllSelected()).toBe(true);
```

Comment:
We should have a test for the inverse of `toggleAll()` to test that it works to unselect, also a test if the locked items are checked too.